src/particle.cpp

```diff
@@ -590,6 +736,7 @@ void Particle::cross_surface()
       return;
     }
   }
+  add_surf_source_to_bank(*this, *surf);
```

Comment:
In progress. I am doing some refactoring to only call the `add_surf_source_to_bank` once.

Reply:
I could not find a good solution to only have one call because in the case where boundary conditions are present, you need to obtain information before handling the condition (i.e., before `cross_surface()` is called). In my last iteration, I call `add_surf_source_to_bank` conditionally twice in `event_cross_surface()` depending on whether a boundary condition is present.